### PR TITLE
Set output encoding properly in LSIF gen scenarios.

### LINKED ISSUE #64050: LSIF Generator can generate invalid JSON in hover results
**Version Used**: `4.4.0-1.22372.4`

**Steps to Reproduce**:

1. Generate LSIF for [this file](https://github.com/dotnet/razor-tooling/blob/0c693f314f2084630fc735379836acadd0ebcc0b/src/Razor/test/Microsoft.AspNetCore.Razor.Test.Common/UseExportProviderAttribute.cs#L33) in dotnet/razor-tooling

**Expected Behavior**:
No 0x07 control character generated

**Actual Behavior**:

Scroll to the right in `value` to see the invalid 0x07 control characters
```JSON
{
	"result": {
		"contents": {
			"kind": "plaintext",
			"value": "class Microsoft.AspNetCore.Razor.Test.Common.UseExportProviderAttribute\r\nThis attribute supports tests that need to use a MEF container (ExportProvider) directly or indirectly during the test sequence. It ensures production code uniformly handles the export provider created during a test, and cleans up the state before the test completes.\r\n\r\nThis attribute serves several important functions for tests that use state variables which are otherwise shared at runtime:\r\n\r\n Ensures HostServices implementations all use the same ExportProvider, which is the one created by the test.\r\n Clears static cached values in production code holding instances of HostServices, or any object obtained from it or one of its related interfaces such as HostLanguageServices.\r\n Isolates tests by waiting for asynchronous operations to complete before a test is considered complete.\r\n When required, provides a separate ExportProvider for the RemoteWorkspace executing in the test process. If this provider is created during testing, it is cleaned up with the primary export provider during test teardown.\r\n"
		},
		"range": {
			"start": {
				"line": 32,
				"character": 17
			},
			"end": {
				"line": 32,
				"character": 43
			}
		}
	},
	"id": 2734,
	"type": "vertex",
	"label": "hoverResult"
}
```

When deserializing this with System.Text.Json you get:
image

And similarly this fails JSON validation on all web validators

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- When we were writing to the console the console was translating characters characters away from their originating type. This resulted in a lot of unexpected JSON failures when deserializing. For instance if you were to have "fancy" Microsoft word quotes in a doc comment the hover results from LSIF generation would contain malformed values because the fancy quote would be translated to a regular quote (breaking the value).

Fixes #64050

### CODE REVIEW

**Comment by NTaylorMullen** [comment]
Oh shoot, I didn't add the output file option fix. That's what I get for submitting this in the middle of a meeting 😄. Will push an update

File: src/Features/Lsif/Generator/Program.cs [hunk 1]
```diff
@@ -43,7 +44,19 @@ private static async Task GenerateAsync(FileInfo? solution, FileInfo? project, F
...
             // If we have an output file, we'll write to that, else we'll use Console.Out
             using var outputFile = output != null ? new StreamWriter(output) : null;
-            var outputWriter = outputFile ?? Console.Out;
+            TextWriter? outputWriter = null;
+
+            if (outputFile is null)
+            {
+                Console.Out.Encoding = UTF8Encoding.UTF8;
+                Console.OutputEncoding = UTF8Encoding.UTF8;
+                Console.Error.Encoding = UTF8Encoding.UTF8;
```
Comment (CyrusNajmabadi) [inline src/Features/Lsif/Generator/Program.cs:53]: interesting.  why do we need to set OutputEncoding and Error.Encoding?  We're just writing to the outputWRiter right?  So i would expect we'd just need the first line.

Can you doc?

File: src/Features/Lsif/Generator/Program.cs [hunk 2]
```diff
@@ -43,7 +44,19 @@ private static async Task GenerateAsync(FileInfo? solution, FileInfo? project, F
         {
             // If we have an output file, we'll write to that, else we'll use Console.Out
             using var outputFile = output != null ? new StreamWriter(output) : null;
-            var outputWriter = outputFile ?? Console.Out;
+            TextWriter? outputWriter = null;
```
Comment (CyrusNajmabadi) [inline src/Features/Lsif/Generator/Program.cs:47]:
```suggestion
            TextWriter? outputWriter;
```

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/Lsif/Generator/Program.cs [hunk 3]
```diff
@@ -43,7 +44,19 @@ private static async Task GenerateAsync(FileInfo? solution, FileInfo? project, F
...
+
+            if (outputFile is null)
+            {
+                Console.Out.Encoding = UTF8Encoding.UTF8;
+                Console.OutputEncoding = UTF8Encoding.UTF8;
```
Comment (jasonmalinowski) [inline src/Features/Lsif/Generator/Program.cs:52]: What's the difference between these two?

File: src/Features/Lsif/Generator/Program.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/Features/Lsif/Generator/Program.cs:53]: Indeed, I wouldn't imagine we're messing with error encodings at all.

**Review by jasonmalinowski — COMMENTED** [review]
Mostly same questions as Cyrus.

File: src/Features/Lsif/Generator/Program.cs (hunk 3 above)
Comment (NTaylorMullen) [reply src/Features/Lsif/Generator/Program.cs:52]: lol, errors were too slow, turns out we only have OutputEncoding available (others are unsettable). Updating

**Comment by NTaylorMullen** [comment]
Updated with the addition to write in UTF8 + changed all the places we set output encoding. Should be good to go 😄

File: src/Features/Lsif/Generator/Program.cs (hunk 1 above)
Comment (NTaylorMullen) [reply src/Features/Lsif/Generator/Program.cs]: Answered below, was a mistake, these aren't actually settable, errors just didn't come up fast enough (compile error)

File: src/Features/Lsif/Generator/Program.cs [hunk 4]
```diff
@@ -42,8 +43,18 @@ public static Task Main(string[] args)
         private static async Task GenerateAsync(FileInfo? solution, FileInfo? project, FileInfo? compilerInvocation, FileInfo? binLog, string? output, LsifFormat outputFormat, string? log)
         {
             // If we have an output file, we'll write to that, else we'll use Console.Out
-            using var outputFile = output != null ? new StreamWriter(output) : null;
-            var outputWriter = outputFile ?? Console.Out;
+            using var outputFile = output != null ? new StreamWriter(output, append: false, Encoding.UTF8) : null;
+            TextWriter? outputWriter;
```
Comment (jasonmalinowski) [inline src/Features/Lsif/Generator/Program.cs:47]:
```suggestion
            TextWriter outputWriter;
```

**Review by jasonmalinowski — APPROVED** [review]

File: src/Features/Lsif/Generator/Program.cs (hunk 4 above)
Comment (NTaylorMullen) [reply src/Features/Lsif/Generator/Program.cs:47]: oh wow, I didn't even know that was allowed

**Pull request merged**